src/main/java/transportservice/ActionListener.java

```diff
@@ -16,8 +16,12 @@ protected static InetSocketAddress getLocalEphemeral() throws UnknownHostExcepti
         return new InetSocketAddress(InetAddress.getLocalHost(), 0);
     }
 
-    public void runActionListener(boolean flag) {
+    public void runActionListener(boolean flag, int timeout) {
         try (ServerSocket socket = new ServerSocket()) {
+
+            // for testing considerations, otherwise zero which is interpreted as an infinite timeout
+            socket.setSoTimeout(timeout);
```

Comment:
Can we use the same timeout here https://github.com/owaiskazi19/IndependentPlugin/blob/44bb779af05732c1326485267722e5f229286376/src/main/java/transportservice/ActionListener.java#L50?